package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,121 @@
+
+def dihedral_calc(atomgroups):
+    """Calculates phi and psi angles for a list of AtomGroups over trajectory.
+
+    Parameters
+    ----------
+    atomgroups : list of AtomGroups
+        must be a list of one or more AtomGroups containing 5 atoms in the
+        correct order (i.e. C-N-CA-C-N)
+
+    Returns
+    -------
+    angles : numpy.ndarray
+        An array of time steps which contain (phi,psi) for all AtomGroups.
+    """
+
+    dihedrals = [atomgroup.dihedral for atomgroup in atomgroups]
+    angles = [dih.value() for dih in dihedrals]
+
+    return angles
+
+
+class Ramachandran(AnalysisBase):
+    """Calculate phi and psi dihedral angles of specified residues.
+
+    Note
+    ----
+    Run the analysis with :meth:`Ramachandran.run()`, which stores the results
+    in the array :attr:`Ramachandran.angles`. A axes object can be obtained
+    with :meth: `Ramachandran.run().plot()`.
+
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the first or last residue
+    then a warning will be raised, and the final array of angles will not
+    include those residues.
+
+    """
+    def __init__(self, atomgroup, **kwargs):
+        r"""Parameters
+        ----------
+        atomgroup : Atomgroup
+            atoms for residues for which phi and psi are calculated
+        start : int, optional
+            starting frame, default None becomes 0.
+        stop : int, optional
+            Frame index to stop analysis. Default: None becomes
+            n_frames. Iteration stops *before* this frame number,
+            which means that the trajectory would be read until the end.
+        step : int, optional
+            step between frames, default None becomes 1.
+
+        """
+        super(Ramachandran, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+
+    def _prepare(self):
+        self.residues = self.atomgroup.residues
+        res_min = np.min(self.atomgroup.universe.select_atoms("protein").residues)
+        res_max = np.max(self.atomgroup.universe.select_atoms("protein").residues)
+        if any([(residue < res_min or residue > res_max) for residue in self.residues]):
+            raise IndexError("Selection exceeds protein length")
+        elif any([residue == (res_min or res_max) for residue in self.residues]):
+            warnings.warn("Cannot determine phi and psi angles for the first or last residues")
+
+        self.phi_atoms = [residue.phi_selection() for residue in self.residues
+                          if res_min < residue < res_max]
+        self.psi_atoms = [residue.psi_selection() for residue in self.residues
+                          if res_min < residue < res_max]
+
+        self.angles = []
+
+    def _single_frame(self):
+        self.phi_angles = dihedral_calc(self.phi_atoms)
+        self.psi_angles = dihedral_calc(self.psi_atoms)
```

Comment:
why keep those around and attach them to self. I would rather make them normal variables that are not added to the class.